src/rgw/rgw_op.cc

```diff
@@ -887,6 +887,20 @@ static bool validate_cors_rule_method(RGWCORSRule *rule, const char *req_meth) {
   return true;
 }
 
+static bool validate_cors_rule_header(RGWCORSRule *rule, const char *req_hdrs) {
+    if (req_hdrs) {
+        list<string> hl;
+        get_str_list(req_hdrs, hl);
+        for(list<string>::iterator it = hl.begin(); it != hl.end(); ++it) {
+            if (!rule->is_header_allowed((*it).c_str(), (*it).length())) {
+                dout(5) << "Header " << (*it) << " is not registered in this rule" << dendl;
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
```

Comment:
looks good, just some nits about style/formatting:
* please use two spaces for indentation,
* prefer std::vector and get_str_vec() to std::list,
* prefer ranged-for loops, i.e. `for (const auto& it : hl) {`